plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -1566,7 +1586,15 @@ private void prepareDropTable(SchemaTableName schemaTableName)
         {
             metastoreDeleteOperations.add(new IrreversibleMetastoreOperation(
                     format("drop table %s", schemaTableName),
-                    () -> delegate.dropTable(schemaTableName.getSchemaName(), schemaTableName.getTableName(), true)));
+                    () -> {
+                        Optional<Table> droppedTable = delegate.getTable(schemaTableName.getSchemaName(), schemaTableName.getTableName());
+                        try {
+                            delegate.dropTable(schemaTableName.getSchemaName(), schemaTableName.getTableName(), true);
+                        }
+                        finally {
+                            droppedTable.ifPresent(tableInvalidationCallback::invalidate);
```

Comment:
in other prepare... methods you call `tablesToInvalidate.add`.
is this different intentionally?

Reply:
Actually it would be consistent with the other changes. I'll add it to `tablesToInvalidate`

Reply:
> The irreversible operations are executed after already calling the invalidation callback

add a code comment.